riscv/runtime/src/coprocessors.rs

```diff
@@ -4,9 +4,9 @@
 // The function itself will be removed by the compiler
 // during the reachability analysis.
 extern "C" {
-    fn poseidon_coprocessor(a: u32, b: u32) -> u32;
+    fn poseidon_gl_coprocessor(data: *mut [u64; 12]);
 }
 
-pub fn poseidon_hash(a: u32, b: u32) -> u32 {
-    unsafe { poseidon_coprocessor(a, b) }
+pub fn poseidon_gl(data: &mut [u64; 12]) {
```

Comment:
Should we add some docstring? If I understand right, it replaces the first 4 elements with the hash, right?